devdoc/object_lifetime_tracker.md

```diff
@@ -151,13 +159,23 @@ MOCKABLE_FUNCTION(, int, object_lifetime_tracker_register_object, OBJECT_LIFETIM
 
   - **SRS_OBJECT_LIFETIME_TRACKER_43_018: [** `object_lifetime_tracker_register_object` shall add the given `key` to the DList of keys by calling `DList_InsertHeadList`. **]**
 
-**SRS_OBJECT_LIFETIME_TRACKER_43_019: [** `object_lifetime_tracker_register_object` shall store the given `object` and the given `destroy_object` in the DList of objects for given `key` by calling `DList_InsertHeadList`. **]**
+**SRS_OBJECT_LIFETIME_TRACKER_43_076: [** `object_lifetime_tracker_register_object` shall find the list entry for the given `object` in the DList of objects for the given `key` by calling `DList_ForEach` with `is_same_object`.**]**
 
-**SRS_OBJECT_LIFETIME_TRACKER_43_051: [** `object_lifetime_tracker_register_object` shall release the lock. **]**
+**SRS_OBJECT_LIFETIME_TRACKER_43_077: [** If the `object` is not found in the DList of objects:**]**
+
+  - **SRS_OBJECT_LIFETIME_TRACKER_43_060: [** `object_lifetime_tracker_register_object` shall allocate memory to store data associated with the `object`. **]**
 
-**SRS_OBJECT_LIFETIME_TRACKER_43_020: [** `object_lifetime_tracker_register_object` shall succeed and return zero. **]**
+  - **SRS_OBJECT_LIFETIME_TRACKER_43_019: [** `object_lifetime_tracker_register_object` shall store the given `object` with the given `destroy_object` and `destroy_context` in the DList of objects for given `key` by calling `DList_InsertHeadList`. **]**
```

Comment:
>* [](http://example.com/codeflow?start=252&length=1)

You can insert in a new line here sort of "Return values:" so the read would know (and not guess) that the following specs are related to "return value from the function".